theurgy.lic

```diff
@@ -1,186 +1,449 @@
-=begin
-  Documentation: https://elanthipedia.play.net/Lich_script_repository#theurgy
-=end
+custom_require.call(%w[common common-arcana common-items common-money common-travel drinfomon spellmonitor equipmanager])
 
-custom_require.call(%w[common common-items common-money common-travel drinfomon spellmonitor equipmanager])
-
-class Theurgy
+class TheurgyActions
   include DRC
   include DRCI
-  include DRCM
   include DRCT
+  include DRCM
+  include DRCA
+  
+  def initialize(settings, data)
+    @settings = settings
+    @hometown = @settings.hometown
+    @data = data
+
+    @theurgy_supply_container = @settings.theurgy_supply_container
+    @water_holder = @settings.water_holder
+    @immortal_aspect = @settings.immortal_aspect
+    @flint_lighter = @settings.flint_lighter
+    @safe_room = @settings.safe_room
+
+    @rituals = get_rituals
+    @communes = get_communes
+
+    annotate_item_presence(@rituals + @communes)
+    
+    @rituals.reject! do |r|
+      r[:items].any? { |i| !i[:on_hand] && !i[:shop] }
+    end
+    @communes.reject! do |c|
+      c[:items].any? { |i| !i[:on_hand] && !i[:shop] }
+    end
+  end
 
-  ######################## GET/PUT ########################
-
-  def put_in_container(item)
-    bput "put #{item} in my #{@theurgy_supply_container}", 'You put', 'What were you referring to'
+  def get_rituals
+    rituals = all_rituals
+    blacklist = @settings.theurgy_blacklist || []
```

Comment:
You can also add the empty array defaults into `base-empty.yaml`

Reply:
Seconded. And for `theurgy_whitelist`.